testing/sliding-sync-integration-test/src/lib.rs

```diff
@@ -900,4 +906,303 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+    async fn growing_sync_keeps_going() -> anyhow::Result<()> {
+        let (_client, sync_proxy_builder) = random_setup_with_rooms(50).await?;
+        let growing_sync = SlidingSyncViewBuilder::default()
+            .sync_mode(SlidingSyncMode::GrowingFullSync)
+            .batch_size(10u32)
+            .sort(vec!["by_recency".to_string(), "by_name".to_string()])
+            .name("growing")
+            .build()?;
+
+        let sync_proxy = sync_proxy_builder.clone().add_view(growing_sync).build().await?;
+        let view = sync_proxy.view("growing").context("but we just added that view!")?;
+
+        let stream = sync_proxy.stream().await?;
+        pin_mut!(stream);
+
+        // we have 50 and catch up in batches of 10. so let's get over to 20.
+
+        for _n in 0..2 {
+            let room_summary = stream.next().await.context("sync has closed unexpectedly")?;
+            let _summary = room_summary?;
+        }
+
+        let collection_simple = view
+            .rooms_list
+            .lock_ref()
+            .iter()
+            .map(Into::<RoomListEntryEasy>::into)
+            .collect::<Vec<_>>();
+        assert_eq!(
+            collection_simple,
+            [
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled, // 10
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled,
+                RoomListEntryEasy::Filled, // 20
+                RoomListEntryEasy::Filled, // 20
```

Comment:
Two times 20?

Reply:
I think something like (`std::iter::`)`repeat(Filled).take(20).chain(repeat(Empty).take(30)).collect()` would be better for readability.